plugins/techdocs-backend/src/techdocs/stages/prepare/gitlab.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import path from 'path';
+import { Entity } from '@backstage/catalog-model';
+import { InputError } from '@backstage/backend-common';
+import { PreparerBase } from './types';
+import parseGitUrl from 'git-url-parse';
+import {
+  parseReferenceAnnotation,
+  checkoutGitRepository,
+} from '../../../helpers';
+
+import { Logger } from 'winston';
+
+export class GitlabPreparer implements PreparerBase {
+  private readonly logger: Logger;
+
+  constructor(logger: Logger) {
+    this.logger = logger;
+  }
+
+  async prepare(entity: Entity): Promise<string> {
+    const { type, target } = parseReferenceAnnotation(
+      'backstage.io/techdocs-ref',
+      entity,
+    );
+
+    if (type !== 'gitlab') {
```

Comment:
If we remove this we could rename this GitPreparer and use that for both gitlab and github. This is a bit of a redundant error.

Reply:
I'm waiting on this PR to start working on Azure DevOps support as well so if this can make that work easier that would be great 😄

Reply:
You want me to do it, or you wish make it after merge?

Reply:
I will make the changes needed for Azure DevOps after this is merged. Was just curious if this would make it easier to implement that.